Sources/App/GithubService+IssueComment.swift

```diff
@@ -0,0 +1,89 @@
+import Vapor
+import Stevenson
+
+struct PingAction: Content {
+    let zen: String
+}
+
+struct CommentAction: Content {
+    let action: String
+    let comment: Comment
+    let issue: Issue
+
+    struct Comment: Content {
+        let body: String
+    }
+
+    struct Issue: Content {
+        let number: Int
+    }
+
+    struct PullRequest: Content {
+        let url: URL
+    }
+}
+
+extension GitHubService {
+    func webhook<T: Decodable>(from request: Request) throws -> Future<T> {
+        let headers = request.http.headers
+
+        guard
+            headers.firstValue(name: .userAgent)?.hasPrefix("GitHub-Hookshot/") == true,
+            headers.firstValue(name: .init("X-GitHub-Event")) == "issue_comment"
+        else {
+            throw Abort(.badRequest)
+        }
+
+        return try request.content.decode(T.self)
+    }
+}
+
+extension GitHubService {
+    // Handle incoming webhook for issue comment
+    func issueComment(
```

Comment:
Just for code reading clarity
```suggestion
    func processComment(
```

Reply:
I'd prefer to keep it as it is